### PR TITLE
fix(components): add accessible label to IconButton

### PR DESCRIPTION
**Pull request by connor-baer**
## Purpose

The `SvgButton` was missing an accessible label for screen readers.

## Approach and changes

- Rename `SvgButton` to `IconButton` (it could be used with an icon font or emoji as well)
- Add a required, visually hidden label to the `IconButton`
- Adjust the usage of the `IconButton`:
  - Simplify the `CloseIcon` component that contained an accessible label
  - Remove the `IconButton` from the `TableHeader` since the icon isn't the clickable element here.

## Definition of done

* [x] Development completed
* [x] Reviewers assigned
* [x] Unit and integration tests
* [x] Meets minimum browser support
* [x] Meets accessibility requirements

### CODE REVIEW

**Comment by lucent1090** [comment]
Thanks for adding me. This was exactly what I need.
by the way, I think { role: 'presentation' } was a good play.

File: src/components/IconButton/IconButton.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      {label && <Label>{label}</Label>}
+    </Button>
+  );
+};
+
+IconButton.propTypes = {
+  /**
+   * The icon element. Must be a single child.
+   */
+  children: PropTypes.element,
```
Comment (herberthenrique) [inline src/components/IconButton/IconButton.js]: Add `isRequired` here :)

**Comment by connor-baer** [comment]
@lucent1090 If you're happy with the changes, would you mind approving the PR, please? :)

**Pull request merged**